app/models/pending_loan.rb

```diff
@@ -10,4 +10,31 @@ def projects
     end
     output
   end
+
+  def calculate_total_cost
+    total_cost = 0
+    projects.each do |project, loan_amount|
+      total_cost += loan_amount.to_i
+    end
+    total_cost
+  end
+
+  def valid?
+    !@pending_loan.empty?
+  end
+
+  def checkout!(user_id)
+    order = Order.create(
+      user_id: user_id.to_i,
+      total_cost: calculate_total_cost,
+      status: 'ordered'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.